### DISCUSSION TITLE
Show sidebar for a doc that isn't assigned to a sidebar

### DISCUSSION THREAD

**Discussion by Dannyholt** [facebook/docusaurus#4728] [Q&A]
We have a lot of WordPress actions and filters we plan to document, but don't want to bloat out the sidebar. Is it possible and if not, can it be considered for adding functionality to show the sidebar for a .md file when it isn't assigned to a sidebar.

For example, we have a .md which lists all filter hooks called Filters Overview. When you click one of the filters lists, we'd like the sidebar to still show the Filters Overview is still highlighted as the active sidebar item. This way we don't have to add over a hundred items to the sidebar. 

I'm not sure how this would work, but it could be that you add some Frontmatter where you select the parent in the example above this would be Filters Overview. 🤔

**Comment by slorber** [comment]
We could allow you to render a given sidebar on a doc that is not in any sidebar by using frontmatter on the doc, like `show_sidebar: filters`

But it would not highlight the Filters Overview page once you navigate to a filter, because you are not on `/filterOverview` anymore but on `/filter/myFilter` and we use the URL for sidebar items highlighting

If you really want the sidebarOverview item of the sidebar to stay highlighted, you'd have to make the filter stay on the same page, perhaps display it in a modal or something? (a modal can be fullscreen).

It is definitively possible to do that with MDX, but I wouldn't recommend it, as it may be better for filters to have unique URLs for deep linking and SEO reasons.

I don't think decorelating the sidebar highlighting from URL would be an easy thing to do

**Reply by Dannyholt** [reply]
Just having the ability to show a sidebar for a doc not assigned to a sidebar would be more than enough. The highlighting doesn't really matter tbh. :)

**Comment by Dannyholt** [comment]
@slorber 

Has there been any movement on this yet. :)

**Reply by slorber** [reply]
Afaik we have a `displayed_sidebar` frontmatter (https://github.com/facebook/docusaurus/pull/5782) to force display a specific sidebar on a doc, but not sure it answers your need

**Comment by Dannyholt** [comment]
@slorber 

Apologizes for the delay, only just around to this now and unfortunately, this doesn't help me. :(

What I'm after is being able to hide a huge list of doc items from the sidebar and simply display an index of those items and when you click on one of the items, it still shows the sidebar that those items are assigned to.

<details><summary>For example:</summary>


<pre><code>
{
         type: 'category',
         label: 'Hooks',
         link: {
            type: 'category',
            type: 'doc',
            id: 'hooks/index',
         },
         items: [
            "hooks/hookA",
            "hooks/hookB",
            "hooks/hookC",
            "hooks/hookD",
            "hooks/hookE",
            "hooks/hookF",
            "hooks/hookG",
         ],
      },
</code></pre>

</details>

When you view the sidebar, all that's listed is the hook/index.md file, which uses the `<DocCardList/>` to display all items assigned to that sidebar.

**Reply by slorber** [reply]
It's really hard for me to figure out what it should look like using text. 

It would help if you used a mockup tool like Excalidraw to show me visually how each URL should render exactly.

**Reply by Dannyholt** [reply]
What I'm after is being able to create a category (folder) and which contains a lot of `.md` files, one for each hook. But instead of it being a sidebar drop-down menu, it acts like an individual doc (think of it as an index) and when you view that doc, it lists all the `.md` files in that category/folder.

The reason for this is we have over 500 hooks, with each hook being an individual doc. When these are added to the sidebar, its a terrible experience as you're scrolling endlessly.

For example, suppose I have a category called **Hooks** and in that category there are over 200 `.md` files for every hook, like the code below:

<details><summary>For example:</summary>


<pre><code>
{
         type: 'category',
         label: 'Hooks',
         link: {
            type: 'category',
            type: 'doc',
            id: 'hooks/index',
         },
         items: [
            "hooks/hookA",
            "hooks/hookB",
            "hooks/hookC",
            "hooks/hookD",
            "hooks/hookE",
            "hooks/hookF",
            "hooks/hookG",
            // and so forth....
         ],
      },
</code></pre>

</details>

I've tried creating the category, adding all the `.md` files to the folder, but not adding them to the sidebar, only listing the `hooks/index` file. But when I use the `<DocCardList />` nothing shows on the index.md file as there isn't anything present in the sidebar.

Hopefully, the image below demonstrates what I am trying to achieve.

<img width="1157" alt="image" src="https://user-images.githubusercontent.com/1708426/221471079-4634c259-ccaf-426d-9913-3b5b4367eaee.png">

Another solution would be to use CSS to hide them from them sidebar using CSS, but there is no easy way to target that particular sidebar drop-down menu. If we could have the ability to add a custom class to specific sidebar items we could use that. For example:

<pre><code>
{
         type: 'category',
         label: 'Hooks',
         link: {
            type: 'category',
            type: 'doc',
            id: 'hooks/index',
            className: 'some-class-name',
         },
         items: [
            "hooks/hookA",
            "hooks/hookB",
            "hooks/hookC",
            "hooks/hookD",
            "hooks/hookE",
            "hooks/hookF",
            "hooks/hookG",
            // and so forth....
         ],
      },
</code></pre>

**Reply by slorber** [reply]
`DocCardList` renders the exact same structure as your sidebar does by default.

It's not documented but you can create a DocCardList yourself if you want to reuse that component with your own tree logic

```jsx
<DocCardList 
  items={[
    sidebarItem1,
    sidebarItem2,
    sidebarCategory1,
    ...
  ]}
/>
```

We don't have a way to compute/store anywhere 2 separate trees. 
But this is probably not needed to achieve your goal: what you want is to have a single tree, but used only in the index page (cards) and avoid displaying the tree on the sidebar.


> Another solution would be to use CSS to hide them from them sidebar using CSS, but there is no easy way to target that particular sidebar drop-down menu. If we could have the ability to add a custom class to specific sidebar items we could use that. 

You can do that:

```js
{
  type: 'category',
  label: 'Hooks',
  className: 'your-custom-class',
  customProps: {yourCustomAttribute: true},
  link: {...}
}
```

The `your-custom-class` will be applied to the sidebar category dom element.

If you swizzle the Category sidebar item component you can also apply your own JSX logic in it, using the `category.customProps.yourCustomAttribute` attribute to decide to not render category items for example.

**Reply by Dannyholt** [reply]
AH, I wasn't aware of `className`. I'll use that. 👍
